ProcessLib/HeatTransportBHE/LocalAssemblers/HeatTransportBHELocalAssemblerSoil-impl.h

```diff
@@ -0,0 +1,129 @@
+{
+template <typename ShapeFunction, typename IntegrationMethod>
+HeatTransportBHELocalAssemblerSoil<ShapeFunction, IntegrationMethod>::
+    HeatTransportBHELocalAssemblerSoil(
+        MeshLib::Element const& e,
+        bool const is_axially_symmetric,
+        unsigned const integration_order,
+        HeatTransportBHEProcessData& process_data)
+    : _process_data(process_data),
+      _integration_method(integration_order),
+      _element_id(e.getID()),
+      _is_axially_symmetric(is_axially_symmetric)
+{
+    unsigned const n_integration_points =
+        _integration_method.getNumberOfPoints();
+
+    _ip_data.reserve(n_integration_points);
+    _secondary_data.N.resize(n_integration_points);
+
+    _shape_matrices = initShapeMatrices<ShapeFunction,
+                                        ShapeMatricesType,
+                                        IntegrationMethod,
+                                        3 /* GlobalDim */>(
+        e, is_axially_symmetric, _integration_method);
+
+    SpatialPosition x_position;
+    x_position.setElementID(_element_id);
+
+    // ip data initialization
+    for (unsigned ip = 0; ip < n_integration_points; ip++)
+    {
+        x_position.setIntegrationPoint(ip);
+
+        // create the class IntegrationPointDataBHE in place
+        auto const& sm = _shape_matrices[ip];
+        double const w = _integration_method.getWeightedPoint(ip).getWeight() *
+                         sm.integralMeasure * sm.detJ;
+        _ip_data.push_back(
+            {sm.N.transpose() * sm.N * w, sm.dNdx.transpose() * sm.dNdx * w});
+
+        _secondary_data.N[ip] = sm.N;
+    }
+}
+
+template <typename ShapeFunction, typename IntegrationMethod>
+void HeatTransportBHELocalAssemblerSoil<ShapeFunction, IntegrationMethod>::
+    assemble(double const t,
+             std::vector<double> const& local_x,
+             std::vector<double>& local_M_data,
+             std::vector<double>& local_K_data,
+             std::vector<double>& /*local_b_data*/)
+{
+    assert(local_x.size() == ShapeFunction::NPOINTS);
+    (void)local_x;  // Avoid unused arg warning.
+
+    auto local_M = MathLib::createZeroedMatrix<NodalMatrixType>(
+        local_M_data, ShapeFunction::NPOINTS, ShapeFunction::NPOINTS);
+    auto local_K = MathLib::createZeroedMatrix<NodalMatrixType>(
+        local_K_data, ShapeFunction::NPOINTS, ShapeFunction::NPOINTS);
+
+    unsigned const n_integration_points =
+        _integration_method.getNumberOfPoints();
+
+    SpatialPosition pos;
+    pos.setElementID(_element_id);
+
+    for (unsigned ip = 0; ip < n_integration_points; ip++)
+    {
+        pos.setIntegrationPoint(ip);
+        auto& ip_data = _ip_data[ip];
+        auto const& M_ip = ip_data.NTN_product_times_w;
+        auto const& K_ip = ip_data.dNdxTdNdx_product_times_w;
+
+        // auto const k_f = _process_data.thermal_conductivity_fluid(t, pos)[0];
+        // auto const k_g = _process_data.thermal_conductivity_gas(t, pos)[0];
```

Comment:
Are k_f and k_g needed for future development?

Reply:
Yes. Currently this is the fully saturated case. I leave the k_f and k_g here in case later on RichardsFlow or MultiphaseFlow process in integrated with the HeatTransportBHE process. For now these parameters are commented out.